src/language-server/universal-data-definition-language-formatter.ts

```diff
@@ -74,21 +119,166 @@ export class UniversalDataDefinitionLanguageFormatter extends AbstractFormatter
+        this.formatContainer(entity)
+        entity.composition.forEach(comp => {
+            this.formatConceptualComposition(comp)
+        })
+    }
+
+    protected formatConceptualAssociation(cassoc: ast.ConceptualAssociation): void {
+        this.formatContainer(cassoc)
+        cassoc.composition.forEach(comp => {
+            this.formatConceptualComposition(comp)
+        })
+    }
+
+    protected formatConceptualComposition(comp: ast.ConceptualComposition   ): void {
+        this.formatContainer(comp)
+    }
+
+    protected formatConceptualParticipant(participant: ast.ConceptualParticipant): void {
+        this.formatContainer(participant)    
+        if(participant?.type){
+            this.formatConceptualEntity(participant.type.ref!)
+        }
+    }
+
+    /**Formatting logicals */
+    protected formatLogicalElement(elem: ast.LogicalElement): void {
+        const formatter = this.getNodeFormatter(elem);
+        formatter.property('name').prepend(Formatting.newLine()).surround(Formatting.oneSpace({allowMore: true}));
+        this.formatObj(elem);   
+    }
+
+    protected formatLogicalEnumerated(lenum: ast.LogicalEnumerated): void {
+        this.formatContainer(lenum)
+        lenum.label.forEach(label => {
+            this.formatContainer(label)
+        })
+    }
+
+    protected formatLogicalMeasurementSystem(sys: ast.LogicalMeasurementSystem): void {
+        this.formatContainer(sys);
+        sys.constraint.forEach(sysConst => {
+            this.formatContainer(sysConst)
+        })
+        sys.referencePoint.forEach(ref => {
+            this.formatLogicalReferencePoint(ref);
+        })
+    }
+
+    protected formatLogicalMeasurementSystemAxis(sysAxis: ast.LogicalMeasurementSystemAxis): void {
+        this.formatContainer(sysAxis);
+        sysAxis.constraint.forEach(sysConst => {
+            this.formatContainer(sysConst);
+        })
+    }
+
+    protected formatLogicalReferencePoint(refPoint: ast.LogicalReferencePoint): void {
+        this.formatContainer(refPoint);
+        refPoint.referencePointPart.forEach(pointPart => {
+            this.formatContainer(pointPart);
+        })
+    }
+
+    protected formatLogicalValueTypeUnit(typeUnit: ast.LogicalValueTypeUnit):void {
+        this.formatContainer(typeUnit);
+        if(typeUnit.constraint){
+            this.formatContainer(typeUnit);
+        }
+    }
+
+    protected formatLogicalMeasurement(measure: ast.LogicalMeasurement): void {
+        this.formatContainer(measure);
+        measure.attribute.forEach(attr => {
+            this.formatContainer(attr)
+        })
+        measure.constraint.forEach(mesConst => {
+            this.formatContainer(mesConst)
+        })
+    }
+
+    protected formatLogicalEntity(entity: ast.LogicalEntity): void {
+        this.formatContainer(entity);
+        entity.composition.forEach(comp => {
+            this.formatContainer(comp);
```

Comment:
Individual composition elements are not containers

Reply:
Thanks for the heads up, will make the changes and revert. I will get back if I am stuck.